modules/core/src/main/java/com/google/refine/model/AbstractOperation.java

```diff
@@ -99,4 +101,23 @@ public String getOperationId() {
     public String getJsonDescription() {
         return getBriefDescription(null);
     }
+
+    /**
+     * A set of columns required by this operation to run. If present, the operation is guaranteed to be able to run on
+     * any project which has at least those columns. If equal to {@link Optional#empty()} the operation could
+     * potentially depend on any column.
+     */
+    @JsonIgnore
+    public Optional<Set<String>> getColumnDependencies() {
+        return Optional.empty();
+    }
+
+    /**
+     * If the effect of the operation on the set of columns in the project is predictable, this effect can be exposed in
+     * this method. Otherwise, {@link Optional#empty()} can be returned.
+     */
+    @JsonIgnore
+    public Optional<ColumnsDiff> getColumnsDiff() {
+        return Optional.empty();
+    }
```

Comment:
Those are the two methods introduced by this PR (with their implementations and tests in a bunch of classes).